### ISSUE TITLE
Improve `manual_strip` fix suggestion

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#14183]
```rust
fn main() {
    let token = "-lfoo";
    if token.starts_with("-l") {
        let libname = &token[2..];
        println!("Library name: {libname}");
    }
}
```

The above code generates this warning with a suggestion:

![Image](https://github.com/user-attachments/assets/7530235f-31fd-4920-bf82-b980490340e1)

Note how the new code uses `<stripped>` instead of a real variable name.  This might be a bit confusing, especially to the new users.  I propose to add a bit more extra logic for this common case:

* if the result is assigned to a new variable, use that binding's name directly in the `Some` expression

This will modify the suggestion from

```rust
if let Some(<stripped>) = token.strip_prefix("-l") {
   let libname = <stripped>;
```

to this

```rust
if let Some(libname) = token.strip_prefix("-l") {
```
